app/src/main/java/de/tum/in/tumcampusapp/service/GeoFencingRegistrationWorker.kt

```diff
@@ -0,0 +1,96 @@
+package de.tum.`in`.tumcampusapp.service
+
+import android.Manifest
+import android.annotation.SuppressLint
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import androidx.core.content.ContextCompat
+import androidx.work.Data
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkRequest
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import com.google.android.gms.location.Geofence
+import com.google.android.gms.location.GeofencingClient
+import com.google.android.gms.location.GeofencingRequest
+import com.google.android.gms.location.LocationServices
+import de.tum.`in`.tumcampusapp.utils.Const
+import de.tum.`in`.tumcampusapp.utils.Utils
+
+class GeoFencingRegistrationWorker (appContext: Context, workerParams: WorkerParameters):
+        Worker(appContext, workerParams) {
+
+    private lateinit var locationClient: GeofencingClient
+
+
+    private fun isLocationPermissionGranted(): Boolean {
+        return ContextCompat.checkSelfPermission(
+                applicationContext, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED
+    }
+
+    @SuppressLint("MissingPermission")
+    override fun doWork(): Result {
+        locationClient = LocationServices.getGeofencingClient(applicationContext)
+
+        Utils.log("Geo fencing service worker started …")
+        if (!isLocationPermissionGranted()) {
+            return Result.failure()
+        }
+        //get the data to create the Geofence request
+        val id = inputData.getString(Const.ADD_GEOFENCE_ID) ?: return Result.failure()
+        val latitude = inputData.getDouble(Const.ADD_GEOFENCE_LAT, 0.0)
+        val longitude = inputData.getDouble(Const.ADD_GEOFENCE_LON, 0.0)
+        val range = inputData.getFloat(Const.ADD_GEOFENCE_RANGE, 0f)
```

Comment:
I think the values which are later used coud be better defaults.
What do you think?

```suggestion
        val latitude = inputData.getDouble(Const.ADD_GEOFENCE_LAT, 48.137430)
        val longitude = inputData.getDouble(Const.ADD_GEOFENCE_LON, 11.575490)
        val range = inputData.getFloat(Const.ADD_GEOFENCE_RANGE, 50_000f)
```

Reply:
Good idea since it's for now the only used numbers by the geofencer. 0 as default value doesn't really make sense in this  situation when I think about it.